crates/examples/src/bin/elfcopy.rs

```diff
@@ -80,8 +105,61 @@ struct DynamicSymbol {
     gnu_hash: Option<u32>,
 }
 
+/// Table that holds a map of the symbols we should rename while copying
+///
+/// This will be loaded by passing a file with lines of the form:
+/// ```
+/// <original_sym> <new_sym>
+/// ```
+/// A symbol name can then be passed to query the corresponding new
+/// name that we should provide the `out_*` variables in `copy_file`.
+struct RedefineSymTable {
+    map: HashMap<String, String>,
```

Comment:
```suggestion
    map: Option<HashMap<Vec<u8>, Vec<u8>>>,
```

This means we don't need any `Option` handling inside `copy_file`, and we don't need any UTF8 conversions.

Reply:
I think I got what you were recommending for this suggestion, but let me know if it needs tweaking.